.github/workflows/build.yml

```diff
@@ -0,0 +1,23 @@
+name: Build sitegen and web site
+
+on: [push]
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+
+    steps:
+    - uses: actions/checkout@v1
+      with:
+        submodules: true
+
+    - name: Setup .NET Core
+      uses: actions/setup-dotnet@v1
+      with:
+        dotnet-version: 3.1.302
+
+    - name: Rebuild sitegen
+      run: make sitegen
+
+    - name: Rebuild site
+      run: make site
```

Comment:
Note: this only builds the site. Publishing is a separate step which hasn't been started yet. I think I'll take that as separate, subsequent PR.